### DISCUSSION TITLE
Something went wrong while executing your query. This may be the result of a timeout

### DISCUSSION THREAD

**Discussion by csantarin** [community/community#24631] [Apps, API and Webhooks]
<pre><code class="lang-auto">Something went wrong while executing your query. This may be the result of a timeout, or it could be a GitHub bug. Please include `0743:3C85:1769D73:3182A23:623AEA61` when reporting this issue.
</code></pre>
<p>Hi. My team’s repo is encountering this error when using the API to create a discussion thread as part of our app release candidate cutoff workflow. The discussion thread contains a list all of the commit hashes of all the PRs merged between the previous and upcoming versions.</p>
<hr>
<p><strong>Here’s the query:</strong></p>
<pre><code class="lang-auto">mutation newDiscussion($repositoryid: ID!, $categoryid: ID!, $body: String!, $title: String! ){
  # input type: CreateDiscussionInput
  createDiscussion(input: {repositoryId: $repositoryid, categoryId: $categoryid, body: $body, title: $title}) {
    # response type: CreateDiscussionPayload
    discussion {
      number
      id
    }
  }
}
</code></pre>
<p><strong>Here’s the input:</strong></p>
<pre><code class="lang-auto">    repositoryid: MDEwOlJlcG9zaXRvcnkxNjIzMTY4MjE=
    categoryid: MDE4OkRpc2N1c3Npb25DYXRlZ29yeTMyNjYxMDQz
    body: "# Note\n
  **To cherry-pick a bug fix to the release branch, kindly comment the PR merge commit hash in this discussion thread.**\n\n
  | ✅ Do this! | ❌ ...and not this. |\n
  | --- | --- |\n
  | ![image](https://user-images.githubusercontent.com/[SOME_NUMBER]/[uuid].png) | ![image](https://user-images.githubusercontent.com/[SOME_NUMBER]/[uuid].png) |\n |\n
  # Pull Requests\n
  [commit-hash] Merge pull request #123 from [Some PR]\n
  [commit-hash] Merge pull request #456 from [Some PR]\n
  [commit-hash] Merge pull request #789 from [Some PR]\n
  ...up to 250 lines of the same commit messages...
</code></pre>
<p>The <code>body</code> field contains a reference to a <code>DISCUSSION_BODY</code> env var which creates the long list of commit messages.</p>
<pre><code class="lang-auto">          BODY=$(git log origin/release/$PREV_RC_VERSION..origin/release/$RC_VERSION  --pretty=tformat:"%h %s\n" --merges --grep="pull request");
          echo 'DISCUSSION_BODY&lt;&lt;EOF' &gt;&gt; $GITHUB_ENV
          echo "$BODY" &gt;&gt; $GITHUB_ENV
          echo 'EOF' &gt;&gt; $GITHUB_ENV
</code></pre>
<p><strong>Here’s the output:</strong><br>
The above error message. The error IDs:</p>
<ul>
<li><code>0743:3C85:1769D73:3182A23:623AEA61</code></li>
<li><code>0403:42BF:3A63D7D:3CB3C24:623AE940</code></li>
</ul>
<p>We ran the same action twice for the same input. It failed at both times.</p>
<hr>
<p>I’d like to point out that the discussion can span several hundred lines (usually about 150), listing out all of the PR merge commits that happened between the previous and next releases. In this particular case, it had close to 250 commit messages.</p>

**Comment by byrneh** [comment]
<p><a class="mention" href="/csantarin">@csantarin</a> as the error says its most likely a timeout due to the complexity and amount of data involved in the query.<br>
If the exact same query works with an example involving less data that would also indicate that as the most likely cause.</p>

**Comment by csantarin** [comment]
<p><a class="mention" href="/byrneh">@byrneh</a> You’re likely on to something.</p>
<p>Our discussion <code>body</code> field looks something like this:</p>
<pre><code class="lang-auto">          body: |
            "# Note\n
            **To cherry-pick a bug fix to the release branch, kindly comment the PR merge commit hash in this discussion thread.**\n\n
            | ✅ Do this! | ❌ ...and not this. |\n
            | --- | --- |\n
            | ![image](https://user-images.githubusercontent.com/[SOME_NUMBER]/[uuid].png) | ![image](https://user-images.githubusercontent.com/[SOME_NUMBER]/[uuid].png) |\n
            # Pull Requests\n"
            ${{ env.DISCUSSION_BODY }}"
            
          title: "${{ github.event.inputs.rc_version_name }}"
</code></pre>
<p>And this <code>env.DISCUSSION_BODY</code> was implemented in a previous step which is pulls all of the changes between two release versions:</p>
<pre><code class="lang-auto">          BODY=$(git log origin/release/$PREV_RC_VERSION..origin/release/$RC_VERSION  --pretty=tformat:"%h %s\n" --merges --grep="pull request");
          echo 'DISCUSSION_BODY&lt;&lt;EOF' &gt;&gt; $GITHUB_ENV
          echo "$BODY" &gt;&gt; $GITHUB_ENV
          echo 'EOF' &gt;&gt; $GITHUB_ENV
</code></pre>
<p>This list alone amounted to <strong>near 250</strong> lines for us at the time. Normally, it has been far less than that (some 150, definitely below 200).</p>
<p>After removing <code>${{ env.DISCUSSION_BODY }}</code> from the <code>body</code> input, the workflow succeeded. There was no timeout.</p>
<p>Is there a known and documented limit to the <code>createDiscussion</code> API method (or just about any GitHub GraphAAQL API method at all)? That will help us understand the limitations better.</p>

**Comment by byrneh** [comment, marked as answer]
<p><a class="mention" href="/csantarin">@csantarin</a> the timeout problem can happen for any graphql that is complex and trying to receive too much data.<br>
Its something I have encountered when trying to perform audits that retrieve lost of data.<br>
When using queries that are at risk of timeout I tune my query to retrieving smaller set of data and leverage paging.<br>
Likely you need to do something similar in terms of breaking up into to chunk sizes that you are confident will not timeout.</p>

**Comment by csantarin** [comment]
<aside class="quote no-group" data-username="byrneh" data-post="4" data-topic="240402">
<div class="title">
<div class="quote-controls"></div>
 byrneh:</div>
<blockquote>
<p>When using queries that are at risk of timeout I tune my query to retrieving smaller set of data and leverage paging.</p>
</blockquote>
</aside>
<p>Pagination, eh? We’ll definitely have a look at that…</p>
<p>My only concern now is how to chain it until all parts of the expected final <code>body</code> have been added to the created discussion.</p>
<p>Perhaps breaking the absurdly large <code>body</code> value off from the <code>createDiscussion</code> call and add a new step with <a href="https://docs.github.com/en/graphql/guides/using-the-graphql-api-for-discussions#updatediscussion" rel="noopener nofollow ugc"><code>updateDiscussion</code></a> can sort of handle this at the side. We only need the ID from the <code>createDiscussion</code> results, anyway. The main content of the discussion can be put afterwards with no need to acknowledge the response… is my take.</p>

**Comment by csantarin** [comment]
<p>GitHub support confirming the replies here so far:</p>
<p>Indeed, we sent a query <code>body</code> that was too long for the API to return a response on time (limit: <strong>10 seconds</strong>). The request waiting time exceeded this limit. As a consequence, we experienced a timeout.</p>
<p>We need to break it down into smaller requests.</p>

**Comment by amber-beasley-liatrio** [comment]
I see this error when using postman or locally trying to make graphql queries against GitHub, but **not** when using the Github graphql explorer:

```
"message": "Something went wrong while executing your query. This may be the result of a timeout, or it could be a GitHub bug. Please include `4A9B:716A:296B9:53EA6:64DCF39F` when reporting this issue."
```

**Comment by CelinePjl** [comment]
Hello,
I'd like to bounce off this thread. I personally have the same error when I try to retrieve PRs that have just been opened on my GitHub repository. The idea is to send notifications on a Slack channel when a PR is opened, merged or closed. The GitHub node is the first one in the scenario

Here is the "details" of the error message 
<img width="400" alt="Capture d’écran 2024-12-05 à 10 20 56" src="https://github.com/user-attachments/assets/f5466f43-5f39-43dc-b00b-9a6416c3cce5">

```
RuntimeError

[502] Something went wrong while executing your query. This may be the result of a timeout, or it could be a GitHub bug. Please include `F072:34648:1A98AD6:1E43BB0:67516E9E` when reporting this issue.
```
My scenario is set up to run every 2 hours on weekdays and only during the day. My workflow **was working at the beginning** but with the time, the **crashes start to appear and to become more and more frequent**. I have no idea how to solve this problem. Could you please help me?
